### DISCUSSION TITLE
Clear widget function doesn't work

### DISCUSSION THREAD

**Discussion by HametAl** [pyvista/pyvista#4025] [Q&A]
I want to add spline or any interactive widget and then clear or at least hid it. When I use clear widget (e.g., p.clear_spline_widget()) the window gets closed and kernel restarts. See below:

```python
import pyvista as pv
import numpy as np

mesh = pv.Wavelet()

# initial spline to seed the example
points = np.array(
    [
        [-8.64208925, -7.34294559, -9.13803458],
        [-8.25601497, -2.54814702, 0.93860914],
        [-0.30179377, -3.21555997, -4.19999019],
        [3.24099167, 2.05814768, 3.39041509],
        [4.39935227, 4.18804542, 8.96391132],
    ]
)

p = pv.Plotter()
p.add_mesh(mesh.outline(), color='black')
p.add_spline_widget(mesh, initial_points=points, n_handles=5)
p.clear_spline_widgets()
p.camera_position = [(30, -42, 30), (0.0, 0.0, 0.0), (-0.09, 0.53, 0.84)]
p.show()
```

**Comment by Autmlif** [comment]
The add_xxx_widget function returns a vtk actor(the widget), you should call its method Off() to shut down before calling clear_xxx_widgets. However, I think this is a design failure for pyvista as Off() is not being included in the clear functon.
